### PR TITLE
Expand debugging services tips

### PR DESCRIPTION
**Pull request by paurullan**
During my first tests I got a lot of trouble for having non-kubernetes compatible Alpine linux images. Also, one time I incorrectly wrote the service selectors.

This PR is just an expansion for those papercuts.

### CODE REVIEW

File: docs/user-guide/debugging-services.md [hunk 1]
```diff
@@ -320,11 +328,17 @@ $ kubectl get service hostnames -o json
...
+Also, check:
+
+* Is the port you are trying to access in `spec.ports[]`?
+* Is the `targetPort` correct for your `Pod`s?
+* If you meant it to be a numeric port, is it a number (9376)
+  or a string "9376"?
+* If you meant it to be a named port, do your `Pod`s expose a port
+  with the same name?
+* Is the port's `protocol` the same as the `Pod`'s?
+* Does the service has a `selector` and is correctly written?
```
Comment (johndmulhausen) [inline docs/user-guide/debugging-services.md]: * Does the service have a selector and is it spelled correctly?

File: docs/user-guide/debugging-services.md [hunk 2]
```diff
@@ -256,6 +256,14 @@ If this fails, you might need to go to the kube-proxy section of this doc, or
 even go back to the top of this document and start over, but instead of
 debugging your own `Service`, debug DNS.
 
+### _Some_ of my images cannot resolve
+
+Check if the container image can resolve DNS via `search`.
+
+For example, Alpine Linux uses `musl-libc` and does not support the `search`
+keyword in `resolv.conf`. If you want to use Alpine as base image you will need
+to get an specially built for Kubernetes.
```
Comment (johndmulhausen) [inline docs/user-guide/debugging-services.md]: If you want to use Alpine as a base image you will need a special build that works with Kubernetes.

File: docs/user-guide/debugging-services.md [hunk 3]
```diff
@@ -256,6 +256,14 @@ If this fails, you might need to go to the kube-proxy section of this doc, or
...
+
```
Comment (johndmulhausen) [inline docs/user-guide/debugging-services.md]: Can you:
- show the `search` command in use?
- link to an Alpine build that works with Kubernetes?

**Comment by bgrant0607** [comment]
cc @kubernetes/sig-network

**Comment by paurullan** [comment]
I signed it!

File: docs/user-guide/debugging-services.md (hunk 2 above)
Comment (thockin) [inline docs/user-guide/debugging-services.md]: This pgph is dated - as of the alpine 3.4, search paths will work.  Can you write this in those terms?

**Comment by paurullan** [comment]
@johndmulhausen I added and example of both questions. Is there anything else to improve?

**Comment by kelseyhightower** [comment]
@paurullan Can you address the issues related to DNS and alpine linux? I'll be happy to review this after that's done. Overall this look pretty good.

**Comment by pwittrock** [comment]
@paurullan Friendly bump.

**Comment by pwittrock** [comment]
@devin-donnelly Would be good to either close this, or merge it and fix the issues as a follow up.  Would you be able to drive this?

**Comment by paurullan** [comment]
@pwittrock @kelseyhightower sorry for the delay, I had a disconnection for the whole August. 
I addressed the comments about the DNS.

**Pull request closed without merging**